src/messaging/ChannelContext.h

```diff
@@ -0,0 +1,153 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the internal classes used by CHIP Channel.
+ */
+
+#pragma once
+
+#include <lib/core/ReferenceCounted.h>
+#include <lib/mdns/platform/Mdns.h>
+#include <messaging/Channel.h>
+#include <transport/CASESession.h>
+#include <transport/PeerConnectionState.h>
+#include <transport/SecureSessionMgr.h>
+
+namespace chip {
+namespace Messaging {
+
+class ExchangeManager;
+class ChannelContext;
+
+class ChannelContextDeletor
+{
+public:
+    static void Release(ChannelContext * context);
+};
+
+class ChannelContext : public ReferenceCounted<ChannelContext, ChannelContextDeletor>, public SessionEstablishmentDelegate
```

Comment:
a brief doc describing the purpose of this class and some sample usage would be helpful.

Reply:
Working on it

Reply:
I'd like to see a state diagram for this.